app/src/main/java/ayds/winchester/songinfo/moredetails/fulllogic/OtherInfoWindow.kt

```
@@ -0,0 +1,213 @@
+        openUrlButton = findViewById(R.id.openUrlButton)
+        dataBase = DataBase(this)
+    }
+
+    private fun open(artist: String) {
+        getArtistInfo(artist)
+    }
+
+    private fun getArtistInfo(artistName: String) {
+        Thread {
+            buildArtistInfo(artistName)
+        }.start()
+    }
+
+    private fun buildArtistInfo(artistName: String) {
+        val artist = searchArtistInfo(artistName)
+        saveArtistInfo(artist)
+        val description = formatArtistInfo(artist)
+        displayWindow(description, artist?.wikipediaURL)
+    }
+
+    private fun displayWindow(description: String,url:String?) {
+        loadWikipediaLogo()
+        updateArtistDescription(description)
+        setButtonUrl(url)
+    }
+
+    private fun formatArtistInfo(artist: WikipediaArtist?): String {
+        return when(artist){
+            is WikipediaArtist ->
+                    (if (artist.isLocallyStored) IS_LOCALLY_STORED_PREFIX else IS_NOT_LOCALLY_STORED_PREFIX) +
+                    formatDescription(artist.description, artist.name)
+
+            else -> NO_RESULTS_MESSAGE
+        }
+    }
+
+    private fun saveArtistInfo(artist: WikipediaArtist?) {
+        artist?.let{
+            dataBase.saveArtist(artist.name, it.description)
+        }
+    }
+
+    private fun formatDescription(description: String, artistName: String): String {
+        val text = description.replace(SLASH_NEW_LINE, NEW_LINE_PLAIN)
+        return textToHtml(text, artistName)
+    }
+
+    private fun searchArtistInfo(artistName: String): WikipediaArtist? {
+        var wikipediaArtist = getArtistFromLocalStorage(artistName)
+        when {
+            wikipediaArtist != null ->  wikipediaArtist.markArtistAsLocal()
+            else -> {
+                try{
+                    wikipediaArtist = getArtistFromWikipedia(artistName)
+                }catch (e1: IOException) {
+                    e1.printStackTrace()
```

no impriman

---

Hola luchi, si sacamos el print nos muestra un warning "Empty catch block", lo sacamos igualmente o lo dejamos?